spec/app/services/scm/checkout_instructions_service_spec.rb

```diff
@@ -0,0 +1,192 @@
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+
+require 'spec_helper'
+
+describe Scm::CheckoutInstructionsService do
+  let(:user) { FactoryGirl.build(:user) }
+  let(:project) { FactoryGirl.build(:project) }
+
+  let(:url) { 'file:///tmp/some/svn/repo' }
+  let(:repository) {
+    FactoryGirl.build(:repository_subversion,
+                      url: url,
+                      project: project)
+  }
+
+  let(:base_url) { 'http://example.org/svn/' }
+  let(:text) { 'foo' }
+  let(:checkout_hash) {
+    {
+      'git' => { 'enabled' => '0' },
+      'subversion' => { 'enabled' => '1',
+                        'text' => text,
+                        'base_url' => base_url
+                      }
+    }
+  }
+
+  subject(:service) { Scm::CheckoutInstructionsService.new(repository, user) }
+
+  before do
+    allow(Setting).to receive(:repository_checkout_data).and_return(checkout_hash)
+  end
+
+  describe '#checkout_url' do
+    shared_examples 'builds the correct URL' do
+      it 'builds the correct URL' do
+        expect(service.checkout_url)
+          .to eq(URI("http://example.org/svn/#{project.identifier}"))
+      end
+
+      it 'builds a subpath' do
+        expect(service.checkout_url('foo/bar'))
+          .to eq(URI("http://example.org/svn/#{project.identifier}/foo/bar"))
+      end
+    end
+
+    it_behaves_like 'builds the correct URL'
+
+    context 'with missing trailing slash' do
+      it_behaves_like 'builds the correct URL'
+    end
+  end
+
+  describe '#checkout_command' do
+    it 'returns the SCM vendor command' do
+      expect(service.checkout_command).to eq('svn checkout')
+    end
+  end
+
+  describe '#instructions' do
+    it 'returns the setting when defined' do
+      expect(service.instructions).to eq('foo')
+    end
+
+    context 'no setting defined' do
+      let(:text) { nil }
+
+      it 'returns the default translated instructions' do
+        expect(service.instructions)
+          .to eq(I18n.t("repositories.checkout.default_instructions.subversion"))
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.